src/Core/Localization/Locale.php

```diff
@@ -149,6 +149,7 @@ public function formatNumber($number)
      */
     public function formatPrice($number, $currencyCode)
     {
+        if ( $number === null ) return null;
```

Comment:
Code style is invalid
Test is inconsistent with expected type of $number argument, phpdoc should be updated to accept `null` type.
Type of returned value is incompatible with method signature.
I suggest this fix instead:
```suggestion
        if ($number === null) {
            return '';
        }
```

Reply:
I'm not sure this is the right approach The method here expects a valid input so it's the calling code which should be careful Besides we are not even sure about what to return in those cases, empty string or null?

That's because this method has no way of knowing how to handle the fallback in this case, only the calling code can know So in the end it will still have to check for the return of this method, check if it's null or empty string and handle the appropriate behaviour

So in both cases, the solution lies in the calling code, so it's preferable to leave this code as is It should fail when it's provided an invalid argument this is the best way to detect unexpected behaviour that are wrong because of the calling code Not the formatting one

Reply:
@jolelievre Then the best solution would be to restrict the input number as `string` only. A simple casting from the calling code would resolve any issue.
I know this would cause a massive BC break, but at least we avoid any confusion.

Reply:
@saulaski an empty string throw an exception too, so casting not fix it 
I think the main question is, formating empty price must be throwing an exception ? 
In this case return exception is a way to detect unexpected behavior or a way to add unexpected behavior ?

Reply:
@mkdgs In this case I agree with @jolelievre, an invalid input to format has to throw an exception.
Number to format can be empty, or simply an invalid number string with unexpected characters. In both cases the value cannot be formatted as a valid number.
The calling code has to use the formatter only when appropriate.

Reply:
@saulaski, @jolelievre 
I think it's excessive for the formatPrice(), it's used for presentational purpose, not validation. 
It's only formating a price, and is not used for calculation.
If formating stuff is needed for validation i think it's a problem.
Imho i think formatPrice() is just a handly method must process empty string and null without any exception.  

If it's definitely not the best choice, I'll be sad.
But I will dry my tears, close this PR and fix that in the calling code.

Reply:
Only a question of point of view.
`formatPrice()` expects a valid value to format as price.
If any value should get formatted as price, or empty string if invalid, then a dedicated method is more appropriate: `formatValueAsPrice()`